### PR TITLE
Documentation of the testing framework

### LINKED ISSUE #143: Testing quarkus github apps documentation
Hi,
I saw you did all the necessary work to allow testing, but the only way to know it is to check the code.
I could also manage to get sample testing code by browsing https://github.com/quarkusio/quarkus-bot. 

I think it could help developers to have all those resources gathered in the documentation.

### PR DESCRIPTION
**Pull request by yrodiere**
Fixes #143.

~First draft, let's talk about it.~ => Ready!

### CODE REVIEW

File: docs/modules/ROOT/pages/testing.adoc [hunk 1]
```diff
@@ -0,0 +1,161 @@
+= Testing
+
+include::./includes/attributes.adoc[]
+
+[[manual-vs-automated]]
+== Manual testing vs. automated testing
+
+The easiest way to start testing your GitHub application is probably
+to create a playground project on GitHub and to run the application locally through Smee.io,
+as explained in xref:register-github-app.adoc[register the application].
```
Comment (gsmet) [inline docs/modules/ROOT/pages/testing.adoc:10]:
```suggestion
as explained in xref:register-github-app.adoc[Register a GitHub App].
```

File: docs/modules/ROOT/pages/testing.adoc [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                            .comment("Hello from my GitHub App"); // <6>
+                });
+    }
+}
+----
+<1> Use `GitHubAppTesting.when()` to start simulating an event.
+<2> Define the payload of the simulated event by pointing to a file in the classpath.
+<3> Define the type of the simulated event.
+<4> Use `.then().github(mocks -> ...)` to perform assertions on GitHub objects involved in the event handling.
+configure mocks.
```
Comment (gsmet) [inline docs/modules/ROOT/pages/testing.adoc:115]: There's something weird here.

**Review by gsmet — CHANGES_REQUESTED** [review]
I spotted minor issues, looks very good otherwise. Thanks!

**Comment by yrodiere** [comment]
Thanks. Should be good now; let's wait for CI.

**Review by gsmet — APPROVED** [review]

**Pull request merged**